### ISSUE TITLE
Validate sites on object instantiation

### ISSUE THREAD

**Issue by ChristopherJHart** [irahorecka/pycraigslist#7]
Is there an easy way to verify if a site (e.g. sfbay, raleigh, etc.) is valid prior to searching via pycraigslist?

Consider the example below:
```python
import pycraigslist

def search_miatas(site: str) -> bool:
    miatas = pycraigslist.forsale.cta(site=site, query="mx5|miata")
    for miata in miatas.search(limit=1):
        print(miata)
```
If I call this function using a known good site, then pycraigslist works as expected.
```
search_miatas("raleigh")

>>> {'country': 'US', 'region': 'NC', 'site': 'raleigh', 'area': '', 'category': 'ctd', 'id': '7358600557', 'repost_of': '7342453821', 'last_updated': '2021-07-30 15:51', 'title': '2016 Mazda MX-5 Miata Grand Touring 2dr Convertible 6M 8000 Miles', 'neighborhood': 'Durham', 'price': '$27,495', 'url': 'https://raleigh.craigslist.org/ctd/d/durham-2016-mazda-mx-miata-grand/7358600557.html'}
```
However, if I call this function with a known incorrect site, then pycraigslist eventually raises MaximumRequestsError.
```
search_miatas("test")

>>> MaximumRequestsError: Maximum requests attempted - check network connection.
```
This makes logical sense, because in the background, pycraigslist is trying to hit `https://test.craigslist.org`, which doesn't exist, so  httpx can't connect, so tenacity raises TryError, which pycraigslist re-raises as MaximumRequestsError.

The problem is that the root cause of MaximumRequestsError could be unclear if I'm accepting site input from an unsanitized source (meaning, raw user input). From what I can tell, there's not an easy way to verify whether MaximumRequestsError is raised because there's a legitimate network connectivity issue, or if the user-defined site is invalid.

I can think of two ways to solve this, assuming it's not solved already:

1. Whenever a pycraigslist.api.BaseAPI object is instantiated, prior to fetching additional filters, parse the [list of Craigslist sites](https://www.craigslist.org/about/sites), validate that the `site` parameter is in the list of sites, and raise an exception if it's not. Downside of this approach is increased execution time (meaning, decreased performance) whenever instantiating a new object.
2. Keep an internal mapping of valid site subdomains and perform the same validation described above. Downside of this approach is, whenever Craigslist adds a new site, you have to throw together a new release to add support for it.

**Comment by irahorecka** [comment]
Hey @ChristopherJHart, thanks for this observation. It appears I cannot get further granularity when distinguishing `requests.exceptions.ConnectionError` from poor network connection and an invalid URL. I like your second suggestion. I don't think Craigslist will be updating their `sites` anytime soon. If they do, I could simply append a new site.

**Comment by irahorecka** [comment]
Should be resolved with v0.6.1 - let me know if this issue persists. I'll go ahead and close this issue.

**Comment by ChristopherJHart** [comment]
Can confirm, this is fixed now. Thank you for the quick turnaround!
